### PR TITLE
Add IP Address to sidebar

### PR DESCRIPTION
**Pull request by TheWizard1328**
- Still needs the variable for actual IP address weather on WiFi or Sim card.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -75,3 +75,5 @@ comma*.sh
 
 selfdrive/modeld/thneed/compile
 models/*.thneed
+/.vs/OpenPilot-jve/v16
```
Comment (j-vanetten) [inline .gitignore]: Keep the `/.vs` and remove `/.vs/OpenPilot-jve/v16` as it's shouldn't be needed

**Review by TheWizard1328 — COMMENTED** [review]
Done ready to go. except for the IP address

File: selfdrive/ui/qt/sidebar.cc [hunk 2]
```diff
@@ -101,11 +101,15 @@ void Sidebar::paintEvent(QPaintEvent *event) {
...
+  p.drawImage(60, 190, signal_imgs[strength]);
   configFont(p, "Open Sans", 35, "Regular");
   p.setPen(QColor(0xff, 0xff, 0xff));
-  const QRect r = QRect(50, 247, 100, 50);
+  const QRect r = QRect(30, 220, 240, 50);
   p.drawText(r, Qt::AlignCenter, network_type[net_type]);
+  // Future line for current IP Address
+  //configFont(p, "Open Sans", 28, "Regular");
+  //const QRect s = QRect(30, 260, 240, 50);
+  //p.drawText(s, Qt::AlignCenter, "000.000.000.000");
```
Comment (j-vanetten) [inline selfdrive/ui/qt/sidebar.cc:112]: Arnepilot has the IP address sent from python.  https://github.com/arne182/ArnePilot/blob/8ed7b03f43d6def9179c862f657498ccaffa5329/selfdrive/ui/sidebar.cc#L67

I'm not sure there is going to be an easy way to get this.  From what I can tell, you'll have to run a shell command to dump the IP address, capture/parse the output and then send it to the UI using the pub/sub messages.

**Comment by TheWizard1328** [comment]
So this would be a no go for addition then???

Wizard Worxx
Where the magic of software begins.
Robert Tauber
(780) 236-1542<tel:(780)%20236-1542>
www.Facebook.com/WizardWorxx13<http://www.facebook.com/WizardWorxx13>


________________________________

**Comment by j-vanetten** [comment]
@TheWizard1328 I personally don't need it so I don't find it useful.  It seems like a lot of overhead just to show the IP address.

**Comment by debugged-tech** [comment]
It's nice to have, but there are other ways to get the IP address and they are not that difficult. Only us dev's usually need to make changes after installed anyway, IMO 🤷🏻‍♂️

**Comment by TheWizard1328** [comment]
I guess so.

Wizard Worxx
Where the magic of software begins.
Robert Tauber
(780) 236-1542<tel:(780)%20236-1542>
www.Facebook.com/WizardWorxx13<http://www.facebook.com/WizardWorxx13>


________________________________

**Comment by j-vanetten** [comment]
While I see the convenience, It's not a complete PR and I think it's just not useful enough to have to maintain.  I do have to keep up with merges from comma's repo and the more changes I add the harder it will be to keep up.

**Comment by TheWizard1328** [comment]
Understandable

Wizard Worxx
Where the magic of software begins.
Robert Tauber
(780) 236-1542<tel:(780)%20236-1542>
www.Facebook.com/WizardWorxx13<http://www.facebook.com/WizardWorxx13>


________________________________

**Pull request closed without merging**